### PR TITLE
Allow setting the name for the base container and fix longstanding bug with running quick pods

### PR DESCRIPTION
**Pull request by csm10495**
Some downstream machinery may require a specific name for the 'base' container.

This change should be backwards compatible since the default base container name is still base.. it can just now be changed to something besides base.

### CODE REVIEW

**Comment by Taragolis** [comment]
Would be nice to have unit tests which cover new functionality for avoid regression

**Comment by csm10495** [comment]
> Would be nice to have unit tests which cover new functionality for avoid regression

Yep.. they're on the way :)

**Comment by csm10495** [comment]
This will also now fix #26796... since i kept hitting it in my tests :) @jedcunningham can you take a look when you get a chance?

File: airflow/providers/cncf/kubernetes/utils/pod_manager.py [hunk 1]
```diff
@@ -265,7 +265,7 @@ def consume_logs(*, since_time: DateTime | None = None, follow: bool = True) ->
                 time.sleep(1)
 
     def await_container_completion(self, pod: V1Pod, container_name: str) -> None:
-        while not self.container_is_running(pod=pod, container_name=container_name):
+        while self.container_is_running(pod=pod, container_name=container_name):
             time.sleep(1)
```
Comment (uranusjr) [inline airflow/providers/cncf/kubernetes/utils/pod_manager.py:269]: While this looks like a valid fix, it is logically unrelated to the `base_container_name` change. Could you split it into another PR?

Comment (csm10495) [reply airflow/providers/cncf/kubernetes/utils/pod_manager.py:269]: The problem is I can't get my new unit tests for base name with get_log=False to pass without fixing this too.

I guess I can split to do this fix first then that one after if you'd prefer.

Comment (uranusjr) [reply airflow/providers/cncf/kubernetes/utils/pod_manager.py:269]: Yeah let’s do that.

Comment (csm10495) [reply airflow/providers/cncf/kubernetes/utils/pod_manager.py:269]: ack. will split em tomorrow.

Comment (csm10495) [reply airflow/providers/cncf/kubernetes/utils/pod_manager.py:269]: @uranusjr i've split this part into: https://github.com/apache/airflow/pull/28771 ... after that merges, i'll recreate this pr.

**Comment by csm10495** [comment]
Closing in favor of #28771 and then redoing this after that.
